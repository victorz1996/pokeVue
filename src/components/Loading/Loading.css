@keyframes rotate {from {transform: rotate(0deg);}
    to {transform: rotate(360deg);}}
@-webkit-keyframes rotate {from {-webkit-transform: rotate(0deg);}
  to {-webkit-transform: rotate(360deg);}}
.carga{
    -webkit-animation: 1s rotate linear infinite;
    animation: 1s rotate linear infinite;
    -webkit-transform-origin: 50% 50%;
    transform-origin: 50% 50%;
}