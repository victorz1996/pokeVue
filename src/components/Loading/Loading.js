/**
 * @namespace Loading
 * @description Componente que se muestra en la pantalla de carga
 * @author       Victor Zabala <victorma164@gmail.com>
 */
export default {
    name: 'Loading',
    props: {
      
    }
}