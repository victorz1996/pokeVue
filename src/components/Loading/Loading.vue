<template>
  <div>
    <div class="row">
      <div class="col-sm-12">
        <img class="carga" src="loading.png" />
      </div>
    </div>
  </div>
</template>
<script src="./Loading.js"></script>
<style lang="css" src="./Loading.css"></style>