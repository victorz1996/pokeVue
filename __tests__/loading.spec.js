import { shallowMount } from '@vue/test-utils'
import Loading from "../src/components/Loading/Loading.vue"

let cmp

    // eslint-disable-next-line no-undef
    beforeEach(() => {
      cmp = shallowMount(Loading)
    })
    // eslint-disable-next-line no-undef
    describe('Componente loading', () => {
      // eslint-disable-next-line no-undef
      test('Instancia de Vue.js',() => {
        // eslint-disable-next-line no-undef
        expect(cmp.isVueInstance).toBeTruthy()
      })

      // eslint-disable-next-line no-undef
      it('HTMl renderizado structure esperada',()=>{
        // eslint-disable-next-line no-undef
        expect(cmp.element).toMatchSnapshot()
    })
})