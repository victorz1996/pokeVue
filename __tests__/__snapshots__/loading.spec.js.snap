// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Componente loading HTMl renderizado structure esperada 1`] = `
<div>
  <div
    class="row"
  >
    <div
      class="col-sm-12"
    >
      <img
        class="carga"
        src="loading.png"
      />
    </div>
  </div>
</div>
`;
